### ISSUE TITLE
pyproject.toml not found when using Poetry in tox' parallel execution

### ISSUE THREAD

**Issue by MaxG87** [tox-dev/tox#1664]
I try to migrate a package that uses Setuptools to Poetry. It turns out that after adapting the configuration of Tox, I cannot longer do parallel executions of my test environments. I asked how to resolve this already on [StackOverflow](https://stackoverflow.com/q/63559236/8831116) but I try to add all relevant information here, of course.

I created a fresh project with no content at all:

```bash
cd /tmp/tox-bug
conda create -yn tox-bug python==3.7 poetry
conda activate tox-bug
poetry new --name tox-bug --src .
poetry add --dev tox
```

Then I add a minimum Tox environment that reproduces the problem:

```
[tox]
envlist = mypy,isort
isolated_build = true
skip_missing_interpreters = False

[testenv:mypy]
allowlist_externals = poetry
commands =
    poetry install -v

[testenv:isort]
allowlist_externals = poetry
commands =
    poetry install -v
```

Of course, in a real project I would actually invoke the linters, but to demonstrate the problem thats not required.

The overall setup is functional. If I just run `tox -rvv`, the outut is:
```
using tox.ini: /tmp/tox-bug/tox.ini (pid 684218)
  removing /tmp/tox-bug/.tox/log
using tox-3.19.0 from $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/__init__.py (pid 684218)
$HOME/.miniconda3/envs/tox-bug/bin/python ($HOME/.miniconda3/envs/tox-bug/bin/python) is {'executable': '$HOME/.miniconda3/envs/tox-bug/bin/python', 'implementation': 'CPython', 'version_info': [3, 7, 0, 'final', 0], 'version': '3.7.0 (default, Oct  9 2018, 10:31:47) \n[GCC 7.3.0]', 'is_64': True, 'sysplatform': 'linux', 'extra_version_info': None}
.package uses $HOME/.miniconda3/envs/tox-bug/bin/python
.package start: getenv /tmp/tox-bug/.tox/.package
.package cannot reuse: -r flag
.package recreate: /tmp/tox-bug/.tox/.package
  removing /tmp/tox-bug/.tox/.package
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684223] /tmp/tox-bug/.tox$ $HOME/.miniconda3/envs/tox-bug/bin/python -m virtualenv --no-download --python $HOME/.miniconda3/envs/tox-bug/bin/python .package
created virtual environment CPython3.7.0.final.0-64 in 116ms
  creator CPython3Posix(dest=/tmp/tox-bug/.tox/.package, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=$HOME/.local/share/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
.package installdeps: poetry>=0.12
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684230] /tmp/tox-bug$ /tmp/tox-bug/.tox/.package/bin/python -m pip install 'poetry>=0.12'
Collecting poetry>=0.12
  Using cached poetry-1.0.10-py2.py3-none-any.whl (224 kB)
Collecting cachecontrol[filecache]<0.13.0,>=0.12.4
  Using cached CacheControl-0.12.6-py2.py3-none-any.whl (19 kB)
Processing $HOME/.cache/pip/wheels/20/78/0a/4d5c19e3868f7dd8df2fa8c2933f01ba28948a3f0b10faccf4/pyrsistent-0.14.11-cp37-cp37m-linux_x86_64.whl
Collecting requests<3.0,>=2.18
  Using cached requests-2.24.0-py2.py3-none-any.whl (61 kB)
Collecting cachy<0.4.0,>=0.3.0
  Using cached cachy-0.3.0-py2.py3-none-any.whl (20 kB)
Collecting cleo<0.8.0,>=0.7.6
  Using cached cleo-0.7.6-py2.py3-none-any.whl (21 kB)
Collecting jsonschema<4.0,>=3.1
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting keyring<21.0.0,>=20.0.1; python_version >= "3.5" and python_version < "4.0"
  Using cached keyring-20.0.1-py2.py3-none-any.whl (31 kB)
Collecting html5lib<2.0,>=1.0
  Using cached html5lib-1.1-py2.py3-none-any.whl (112 kB)
Collecting requests-toolbelt<0.9.0,>=0.8.0
  Using cached requests_toolbelt-0.8.0-py2.py3-none-any.whl (54 kB)
Collecting pyparsing<3.0,>=2.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting clikit<0.5.0,>=0.4.2
  Using cached clikit-0.4.3-py2.py3-none-any.whl (88 kB)
Collecting pexpect<5.0.0,>=4.7.0
  Using cached pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting shellingham<2.0,>=1.1
  Using cached shellingham-1.3.2-py2.py3-none-any.whl (11 kB)
Collecting tomlkit<0.6.0,>=0.5.11
  Using cached tomlkit-0.5.11-py2.py3-none-any.whl (31 kB)
Collecting pkginfo<2.0,>=1.4
  Using cached pkginfo-1.5.0.1-py2.py3-none-any.whl (25 kB)
Collecting importlib-metadata<1.2.0,>=1.1.3; python_version < "3.8"
  Using cached importlib_metadata-1.1.3-py2.py3-none-any.whl (29 kB)
Collecting msgpack>=0.5.2
  Using cached msgpack-1.0.0-cp37-cp37m-manylinux1_x86_64.whl (275 kB)
Collecting lockfile>=0.9; extra == "filecache"
  Using cached lockfile-0.12.2-py2.py3-none-any.whl (13 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2020.6.20-py2.py3-none-any.whl (156 kB)
Collecting chardet<4,>=3.0.2
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting idna<3,>=2.5
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Using cached urllib3-1.25.10-py2.py3-none-any.whl (127 kB)
Collecting attrs>=17.4.0
  Using cached attrs-20.1.0-py2.py3-none-any.whl (49 kB)
Requirement already satisfied: setuptools in ./.tox/.package/lib/python3.7/site-packages (from jsonschema<4.0,>=3.1->poetry>=0.12) (49.6.0)
Collecting secretstorage; sys_platform == "linux"
  Using cached SecretStorage-3.1.2-py3-none-any.whl (14 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting pylev<2.0,>=1.3
  Using cached pylev-1.3.0-py2.py3-none-any.whl (4.9 kB)
Collecting pastel<0.3.0,>=0.2.0
  Using cached pastel-0.2.0-py2.py3-none-any.whl (6.0 kB)
Collecting ptyprocess>=0.5
  Using cached ptyprocess-0.6.0-py2.py3-none-any.whl (39 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
Collecting jeepney>=0.4.2
  Using cached jeepney-0.4.3-py3-none-any.whl (21 kB)
Collecting cryptography
  Using cached cryptography-3.1-cp35-abi3-manylinux2010_x86_64.whl (2.6 MB)
Collecting cffi!=1.11.3,>=1.8
  Using cached cffi-1.14.2-cp37-cp37m-manylinux1_x86_64.whl (401 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Installing collected packages: certifi, chardet, idna, urllib3, requests, msgpack, lockfile, cachecontrol, six, pyrsistent, cachy, pylev, pastel, clikit, cleo, attrs, zipp, importlib-metadata, jsonschema, jeepney, pycparser, cffi, cryptography, secretstorage, keyring, webencodings, html5lib, requests-toolbelt, pyparsing, ptyprocess, pexpect, shellingham, tomlkit, pkginfo, poetry
Successfully installed attrs-20.1.0 cachecontrol-0.12.6 cachy-0.3.0 certifi-2020.6.20 cffi-1.14.2 chardet-3.0.4 cleo-0.7.6 clikit-0.4.3 cryptography-3.1 html5lib-1.1 idna-2.10 importlib-metadata-1.1.3 jeepney-0.4.3 jsonschema-3.2.0 keyring-20.0.1 lockfile-0.12.2 msgpack-1.0.0 pastel-0.2.0 pexpect-4.8.0 pkginfo-1.5.0.1 poetry-1.0.10 ptyprocess-0.6.0 pycparser-2.20 pylev-1.3.0 pyparsing-2.4.7 pyrsistent-0.14.11 requests-2.24.0 requests-toolbelt-0.8.0 secretstorage-3.1.2 shellingham-1.3.2 six-1.15.0 tomlkit-0.5.11 urllib3-1.25.10 webencodings-0.5.1 zipp-3.1.0
.package finish: getenv /tmp/tox-bug/.tox/.package after 6.65 seconds
.package start: finishvenv 
write config to /tmp/tox-bug/.tox/.package/.tox-config1 as 'afa2b369944ea1420a154409f1b3b60f047f01818080b100652e6d86fa4f0126 $HOME/.miniconda3/envs/tox-bug/bin/python\n3.19.0 0 0 0\n00000000000000000000000000000000 poetry>=0.12'
.package finish: finishvenv  after 0.01 seconds
.package start: get-build-requires /tmp/tox-bug/.tox/.package
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684326] /tmp/tox-bug$ /tmp/tox-bug/.tox/.package/bin/python $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/helper/build_requires.py poetry.masonry.api '' >.tox/.package/log/.package-0.log
.package finish: get-build-requires /tmp/tox-bug/.tox/.package after 0.28 seconds
.package start: perform-isolated-build /tmp/tox-bug/.tox/.package
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684327] /tmp/tox-bug$ /tmp/tox-bug/.tox/.package/bin/python $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/helper/build_isolated.py .tox/dist poetry.masonry.api '' '' >.tox/.package/log/.package-1.log
tox-bug-0.1.0.tar.gz

.package finish: perform-isolated-build /tmp/tox-bug/.tox/.package after 0.31 seconds
copying new sdistfile to '$HOME/.tox/distshare/tox-bug-0.1.0.tar.gz'
package .tmp/package/1/tox-bug-0.1.0.tar.gz links to dist/tox-bug-0.1.0.tar.gz (/tmp/tox-bug/.tox)
mypy uses $HOME/.miniconda3/envs/tox-bug/bin/python
mypy start: getenv /tmp/tox-bug/.tox/mypy
mypy cannot reuse: -r flag
mypy recreate: /tmp/tox-bug/.tox/mypy
  removing /tmp/tox-bug/.tox/mypy
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684330] /tmp/tox-bug/.tox$ $HOME/.miniconda3/envs/tox-bug/bin/python -m virtualenv --no-download --python $HOME/.miniconda3/envs/tox-bug/bin/python mypy
created virtual environment CPython3.7.0.final.0-64 in 120ms
  creator CPython3Posix(dest=/tmp/tox-bug/.tox/mypy, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=$HOME/.local/share/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
mypy finish: getenv /tmp/tox-bug/.tox/mypy after 0.30 seconds
mypy start: installpkg /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
mypy inst: /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
write config to /tmp/tox-bug/.tox/mypy/.tox-config1 as 'afa2b369944ea1420a154409f1b3b60f047f01818080b100652e6d86fa4f0126 $HOME/.miniconda3/envs/tox-bug/bin/python\n3.19.0 0 0 0'
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684338] /tmp/tox-bug$ /tmp/tox-bug/.tox/mypy/bin/python -m pip install --exists-action w .tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
Processing ./.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: tox-bug
  Building wheel for tox-bug (PEP 517): started
  Building wheel for tox-bug (PEP 517): finished with status 'done'
  Created wheel for tox-bug: filename=tox_bug-0.1.0-py3-none-any.whl size=1020 sha256=5d24828056fb6b4f155f98923bf14c8554f9d3d4c2052715bcb1659f3159c5c9
  Stored in directory: $HOME/.cache/pip/wheels/5f/8d/8d/56613c7f4eba1b0c0b0dc2928e498e7b7d3a5e32661e5be64f
Successfully built tox-bug
Installing collected packages: tox-bug
Successfully installed tox-bug-0.1.0
mypy finish: installpkg /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz after 8.08 seconds
mypy start: envreport 
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684514] /tmp/tox-bug$ /tmp/tox-bug/.tox/mypy/bin/python -m pip freeze >.tox/mypy/log/mypy-0.log
mypy finish: envreport  after 0.25 seconds
mypy installed: tox-bug @ file:///tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
mypy start: run-test-pre 
mypy run-test-pre: PYTHONHASHSEED='193803339'
mypy finish: run-test-pre  after 0.00 seconds
mypy start: run-test 
mypy run-test: commands[0] | poetry install -v
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684515] /tmp/tox-bug$ $HOME/.miniconda3/envs/tox-bug/bin/poetry install -v
Using virtualenv: /tmp/tox-bug/.tox/mypy
Installing dependencies from lock file


Package operations: 17 installs, 0 updates, 0 removals, 2 skipped

  - Installing zipp (3.1.0)
  - Installing appdirs (1.4.4)
  - Installing distlib (0.3.1)
  - Installing filelock (3.0.12)
  - Installing importlib-metadata (1.7.0)
  - Installing pyparsing (2.4.7)
  - Installing six (1.15.0)
  - Skipping atomicwrites (1.4.0) Not needed for the current environment
  - Installing attrs (20.1.0)
  - Skipping colorama (0.4.3) Not needed for the current environment
  - Installing more-itertools (8.4.0)
  - Installing packaging (20.4)
  - Installing pluggy (0.13.1)
  - Installing py (1.9.0)
  - Installing toml (0.10.1)
  - Installing virtualenv (20.0.31)
  - Installing wcwidth (0.2.5)
  - Installing pytest (5.4.3)
  - Installing tox (3.19.0)
  - Installing tox-bug (0.1.0)
mypy finish: run-test  after 12.89 seconds
mypy start: run-test-post 
mypy finish: run-test-post  after 0.00 seconds
isort uses $HOME/.miniconda3/envs/tox-bug/bin/python
isort start: getenv /tmp/tox-bug/.tox/isort
isort cannot reuse: -r flag
isort recreate: /tmp/tox-bug/.tox/isort
  removing /tmp/tox-bug/.tox/isort
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684853] /tmp/tox-bug/.tox$ $HOME/.miniconda3/envs/tox-bug/bin/python -m virtualenv --no-download --python $HOME/.miniconda3/envs/tox-bug/bin/python isort
created virtual environment CPython3.7.0.final.0-64 in 170ms
  creator CPython3Posix(dest=/tmp/tox-bug/.tox/isort, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=$HOME/.local/share/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
isort finish: getenv /tmp/tox-bug/.tox/isort after 0.39 seconds
isort start: installpkg /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
isort inst: /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
write config to /tmp/tox-bug/.tox/isort/.tox-config1 as 'afa2b369944ea1420a154409f1b3b60f047f01818080b100652e6d86fa4f0126 $HOME/.miniconda3/envs/tox-bug/bin/python\n3.19.0 0 0 0'
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[684903] /tmp/tox-bug$ /tmp/tox-bug/.tox/isort/bin/python -m pip install --exists-action w .tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
Processing ./.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: tox-bug
  Building wheel for tox-bug (PEP 517): started
  Building wheel for tox-bug (PEP 517): finished with status 'done'
  Created wheel for tox-bug: filename=tox_bug-0.1.0-py3-none-any.whl size=1020 sha256=5d24828056fb6b4f155f98923bf14c8554f9d3d4c2052715bcb1659f3159c5c9
  Stored in directory: $HOME/.cache/pip/wheels/5f/8d/8d/56613c7f4eba1b0c0b0dc2928e498e7b7d3a5e32661e5be64f
Successfully built tox-bug
Installing collected packages: tox-bug
Successfully installed tox-bug-0.1.0
isort finish: installpkg /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz after 8.16 seconds
isort start: envreport 
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[685006] /tmp/tox-bug$ /tmp/tox-bug/.tox/isort/bin/python -m pip freeze >.tox/isort/log/isort-0.log
isort finish: envreport  after 0.26 seconds
isort installed: tox-bug @ file:///tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
isort start: run-test-pre 
isort run-test-pre: PYTHONHASHSEED='193803339'
isort finish: run-test-pre  after 0.00 seconds
isort start: run-test 
isort run-test: commands[0] | poetry install -v
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[685007] /tmp/tox-bug$ $HOME/.miniconda3/envs/tox-bug/bin/poetry install -v
Using virtualenv: /tmp/tox-bug/.tox/isort
Installing dependencies from lock file


Package operations: 17 installs, 0 updates, 0 removals, 2 skipped

  - Installing zipp (3.1.0)
  - Installing appdirs (1.4.4)
  - Installing distlib (0.3.1)
  - Installing filelock (3.0.12)
  - Installing importlib-metadata (1.7.0)
  - Installing pyparsing (2.4.7)
  - Installing six (1.15.0)
  - Skipping atomicwrites (1.4.0) Not needed for the current environment
  - Installing attrs (20.1.0)
  - Skipping colorama (0.4.3) Not needed for the current environment
  - Installing more-itertools (8.4.0)
  - Installing packaging (20.4)
  - Installing pluggy (0.13.1)
  - Installing py (1.9.0)
  - Installing toml (0.10.1)
  - Installing virtualenv (20.0.31)
  - Installing wcwidth (0.2.5)
  - Installing pytest (5.4.3)
  - Installing tox (3.19.0)
  - Installing tox-bug (0.1.0)
isort finish: run-test  after 12.80 seconds
isort start: run-test-post 
isort finish: run-test-post  after 0.00 seconds
___________________________________ summary ____________________________________
  mypy: commands succeeded
  isort: commands succeeded
  congratulations :)
cleanup /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
```

However, if I run the environments in parallel with `tox -rvv --parallel auto --parallel-live`, one of the environments will fail because it is missing `pyproject.toml`. The output is:

```
using tox.ini: /tmp/tox-bug/tox.ini (pid 697071)
  removing /tmp/tox-bug/.tox/log
using tox-3.19.0 from $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/__init__.py (pid 697071)
$HOME/.miniconda3/envs/tox-bug/bin/python ($HOME/.miniconda3/envs/tox-bug/bin/python) is {'executable': '$HOME/.miniconda3/envs/tox-bug/bin/python', 'implementation': 'CPython', 'version_info': [3, 7, 0, 'final', 0], 'version': '3.7.0 (default, Oct  9 2018, 10:31:47) \n[GCC 7.3.0]', 'is_64': True, 'sysplatform': 'linux', 'extra_version_info': None}
.package uses $HOME/.miniconda3/envs/tox-bug/bin/python
.package start: getenv /tmp/tox-bug/.tox/.package
.package cannot reuse: -r flag
.package recreate: /tmp/tox-bug/.tox/.package
  removing /tmp/tox-bug/.tox/.package
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697076] /tmp/tox-bug/.tox$ $HOME/.miniconda3/envs/tox-bug/bin/python -m virtualenv --no-download --python $HOME/.miniconda3/envs/tox-bug/bin/python .package
created virtual environment CPython3.7.0.final.0-64 in 137ms
  creator CPython3Posix(dest=/tmp/tox-bug/.tox/.package, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=$HOME/.local/share/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
.package installdeps: poetry>=0.12
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697112] /tmp/tox-bug$ /tmp/tox-bug/.tox/.package/bin/python -m pip install 'poetry>=0.12'
Collecting poetry>=0.12
  Using cached poetry-1.0.10-py2.py3-none-any.whl (224 kB)
Processing $HOME/.cache/pip/wheels/20/78/0a/4d5c19e3868f7dd8df2fa8c2933f01ba28948a3f0b10faccf4/pyrsistent-0.14.11-cp37-cp37m-linux_x86_64.whl
Collecting clikit<0.5.0,>=0.4.2
  Using cached clikit-0.4.3-py2.py3-none-any.whl (88 kB)
Collecting requests-toolbelt<0.9.0,>=0.8.0
  Using cached requests_toolbelt-0.8.0-py2.py3-none-any.whl (54 kB)
Collecting cleo<0.8.0,>=0.7.6
  Using cached cleo-0.7.6-py2.py3-none-any.whl (21 kB)
Collecting requests<3.0,>=2.18
  Using cached requests-2.24.0-py2.py3-none-any.whl (61 kB)
Collecting importlib-metadata<1.2.0,>=1.1.3; python_version < "3.8"
  Using cached importlib_metadata-1.1.3-py2.py3-none-any.whl (29 kB)
Collecting keyring<21.0.0,>=20.0.1; python_version >= "3.5" and python_version < "4.0"
  Using cached keyring-20.0.1-py2.py3-none-any.whl (31 kB)
Collecting tomlkit<0.6.0,>=0.5.11
  Using cached tomlkit-0.5.11-py2.py3-none-any.whl (31 kB)
Collecting pyparsing<3.0,>=2.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting pexpect<5.0.0,>=4.7.0
  Using cached pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting html5lib<2.0,>=1.0
  Using cached html5lib-1.1-py2.py3-none-any.whl (112 kB)
Collecting jsonschema<4.0,>=3.1
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting pkginfo<2.0,>=1.4
  Using cached pkginfo-1.5.0.1-py2.py3-none-any.whl (25 kB)
Collecting cachy<0.4.0,>=0.3.0
  Using cached cachy-0.3.0-py2.py3-none-any.whl (20 kB)
Collecting cachecontrol[filecache]<0.13.0,>=0.12.4
  Using cached CacheControl-0.12.6-py2.py3-none-any.whl (19 kB)
Collecting shellingham<2.0,>=1.1
  Using cached shellingham-1.3.2-py2.py3-none-any.whl (11 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting pastel<0.3.0,>=0.2.0
  Using cached pastel-0.2.0-py2.py3-none-any.whl (6.0 kB)
Collecting pylev<2.0,>=1.3
  Using cached pylev-1.3.0-py2.py3-none-any.whl (4.9 kB)
Collecting idna<3,>=2.5
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2020.6.20-py2.py3-none-any.whl (156 kB)
Collecting chardet<4,>=3.0.2
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Using cached urllib3-1.25.10-py2.py3-none-any.whl (127 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
Collecting secretstorage; sys_platform == "linux"
  Using cached SecretStorage-3.1.2-py3-none-any.whl (14 kB)
Collecting ptyprocess>=0.5
  Using cached ptyprocess-0.6.0-py2.py3-none-any.whl (39 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting attrs>=17.4.0
  Using cached attrs-20.1.0-py2.py3-none-any.whl (49 kB)
Requirement already satisfied: setuptools in ./.tox/.package/lib/python3.7/site-packages (from jsonschema<4.0,>=3.1->poetry>=0.12) (49.6.0)
Collecting msgpack>=0.5.2
  Using cached msgpack-1.0.0-cp37-cp37m-manylinux1_x86_64.whl (275 kB)
Collecting lockfile>=0.9; extra == "filecache"
  Using cached lockfile-0.12.2-py2.py3-none-any.whl (13 kB)
Collecting jeepney>=0.4.2
  Using cached jeepney-0.4.3-py3-none-any.whl (21 kB)
Collecting cryptography
  Using cached cryptography-3.1-cp35-abi3-manylinux2010_x86_64.whl (2.6 MB)
Collecting cffi!=1.11.3,>=1.8
  Using cached cffi-1.14.2-cp37-cp37m-manylinux1_x86_64.whl (401 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Installing collected packages: six, pyrsistent, pastel, pylev, clikit, idna, certifi, chardet, urllib3, requests, requests-toolbelt, cleo, zipp, importlib-metadata, jeepney, pycparser, cffi, cryptography, secretstorage, keyring, tomlkit, pyparsing, ptyprocess, pexpect, webencodings, html5lib, attrs, jsonschema, pkginfo, cachy, msgpack, lockfile, cachecontrol, shellingham, poetry
Successfully installed attrs-20.1.0 cachecontrol-0.12.6 cachy-0.3.0 certifi-2020.6.20 cffi-1.14.2 chardet-3.0.4 cleo-0.7.6 clikit-0.4.3 cryptography-3.1 html5lib-1.1 idna-2.10 importlib-metadata-1.1.3 jeepney-0.4.3 jsonschema-3.2.0 keyring-20.0.1 lockfile-0.12.2 msgpack-1.0.0 pastel-0.2.0 pexpect-4.8.0 pkginfo-1.5.0.1 poetry-1.0.10 ptyprocess-0.6.0 pycparser-2.20 pylev-1.3.0 pyparsing-2.4.7 pyrsistent-0.14.11 requests-2.24.0 requests-toolbelt-0.8.0 secretstorage-3.1.2 shellingham-1.3.2 six-1.15.0 tomlkit-0.5.11 urllib3-1.25.10 webencodings-0.5.1 zipp-3.1.0
.package finish: getenv /tmp/tox-bug/.tox/.package after 6.13 seconds
.package start: finishvenv 
write config to /tmp/tox-bug/.tox/.package/.tox-config1 as 'afa2b369944ea1420a154409f1b3b60f047f01818080b100652e6d86fa4f0126 $HOME/.miniconda3/envs/tox-bug/bin/python\n3.19.0 0 0 0\n00000000000000000000000000000000 poetry>=0.12'
.package finish: finishvenv  after 0.01 seconds
.package start: get-build-requires /tmp/tox-bug/.tox/.package
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697261] /tmp/tox-bug$ /tmp/tox-bug/.tox/.package/bin/python $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/helper/build_requires.py poetry.masonry.api '' >.tox/.package/log/.package-0.log
.package finish: get-build-requires /tmp/tox-bug/.tox/.package after 0.30 seconds
.package start: perform-isolated-build /tmp/tox-bug/.tox/.package
setting PATH=/tmp/tox-bug/.tox/.package/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697262] /tmp/tox-bug$ /tmp/tox-bug/.tox/.package/bin/python $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/helper/build_isolated.py .tox/dist poetry.masonry.api '' '' >.tox/.package/log/.package-1.log
tox-bug-0.1.0.tar.gz

.package finish: perform-isolated-build /tmp/tox-bug/.tox/.package after 0.31 seconds
copying new sdistfile to '$HOME/.tox/distshare/tox-bug-0.1.0.tar.gz'
package .tmp/package/1/tox-bug-0.1.0.tar.gz links to dist/tox-bug-0.1.0.tar.gz (/tmp/tox-bug/.tox)
mypy uses $HOME/.miniconda3/envs/tox-bug/bin/python
mypy start: parallel mypy 
isort uses $HOME/.miniconda3/envs/tox-bug/bin/python
isort start: parallel isort 
[697267] /tmp/tox-bug$ $HOME/.miniconda3/envs/tox-bug/bin/python $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/__main__.py -rvv --parallel auto --parallel-live --installpkg .tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
[697268] /tmp/tox-bug$ $HOME/.miniconda3/envs/tox-bug/bin/python $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/__main__.py -rvv --parallel auto --parallel-live --installpkg .tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
using tox.ini: /tmp/tox-bug/tox.ini (pid 697267)
using tox-3.19.0 from $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/__init__.py (pid 697267)
using tox.ini: /tmp/tox-bug/tox.ini (pid 697268)
using package '/tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz', skipping 'sdist' activity 
package 2/tox-bug-0.1.0.tar.gz links to 1/tox-bug-0.1.0.tar.gz (/tmp/tox-bug/.tox/.tmp/package)
using tox-3.19.0 from $HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/tox/__init__.py (pid 697268)
using package '/tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz', skipping 'sdist' activity 
package 3/tox-bug-0.1.0.tar.gz links to 1/tox-bug-0.1.0.tar.gz (/tmp/tox-bug/.tox/.tmp/package)
$HOME/.miniconda3/envs/tox-bug/bin/python ($HOME/.miniconda3/envs/tox-bug/bin/python) is {'executable': '$HOME/.miniconda3/envs/tox-bug/bin/python', 'implementation': 'CPython', 'version_info': [3, 7, 0, 'final', 0], 'version': '3.7.0 (default, Oct  9 2018, 10:31:47) \n[GCC 7.3.0]', 'is_64': True, 'sysplatform': 'linux', 'extra_version_info': None}
mypy uses $HOME/.miniconda3/envs/tox-bug/bin/python
mypy start: getenv /tmp/tox-bug/.tox/mypy
mypy cannot reuse: -r flag
mypy recreate: /tmp/tox-bug/.tox/mypy
$HOME/.miniconda3/envs/tox-bug/bin/python ($HOME/.miniconda3/envs/tox-bug/bin/python) is {'executable': '$HOME/.miniconda3/envs/tox-bug/bin/python', 'implementation': 'CPython', 'version_info': [3, 7, 0, 'final', 0], 'version': '3.7.0 (default, Oct  9 2018, 10:31:47) \n[GCC 7.3.0]', 'is_64': True, 'sysplatform': 'linux', 'extra_version_info': None}
isort uses $HOME/.miniconda3/envs/tox-bug/bin/python
isort start: getenv /tmp/tox-bug/.tox/isort
isort cannot reuse: -r flag
isort recreate: /tmp/tox-bug/.tox/isort
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697275] /tmp/tox-bug/.tox$ $HOME/.miniconda3/envs/tox-bug/bin/python -m virtualenv --no-download --python $HOME/.miniconda3/envs/tox-bug/bin/python mypy
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697276] /tmp/tox-bug/.tox$ $HOME/.miniconda3/envs/tox-bug/bin/python -m virtualenv --no-download --python $HOME/.miniconda3/envs/tox-bug/bin/python isort
created virtual environment CPython3.7.0.final.0-64 in 124ms
  creator CPython3Posix(dest=/tmp/tox-bug/.tox/mypy, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=$HOME/.local/share/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
created virtual environment CPython3.7.0.final.0-64 in 136ms
  creator CPython3Posix(dest=/tmp/tox-bug/.tox/isort, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=$HOME/.local/share/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
mypy finish: getenv /tmp/tox-bug/.tox/mypy after 0.33 seconds
mypy start: installpkg /tmp/tox-bug/.tox/.tmp/package/2/tox-bug-0.1.0.tar.gz
mypy inst: /tmp/tox-bug/.tox/.tmp/package/2/tox-bug-0.1.0.tar.gz
write config to /tmp/tox-bug/.tox/mypy/.tox-config1 as 'afa2b369944ea1420a154409f1b3b60f047f01818080b100652e6d86fa4f0126 $HOME/.miniconda3/envs/tox-bug/bin/python\n3.19.0 0 0 0'
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697291] /tmp/tox-bug$ /tmp/tox-bug/.tox/mypy/bin/python -m pip install --exists-action w .tox/.tmp/package/2/tox-bug-0.1.0.tar.gz
isort finish: getenv /tmp/tox-bug/.tox/isort after 0.35 seconds
isort start: installpkg /tmp/tox-bug/.tox/.tmp/package/3/tox-bug-0.1.0.tar.gz
isort inst: /tmp/tox-bug/.tox/.tmp/package/3/tox-bug-0.1.0.tar.gz
write config to /tmp/tox-bug/.tox/isort/.tox-config1 as 'afa2b369944ea1420a154409f1b3b60f047f01818080b100652e6d86fa4f0126 $HOME/.miniconda3/envs/tox-bug/bin/python\n3.19.0 0 0 0'
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697292] /tmp/tox-bug$ /tmp/tox-bug/.tox/isort/bin/python -m pip install --exists-action w .tox/.tmp/package/3/tox-bug-0.1.0.tar.gz
Processing ./.tox/.tmp/package/2/tox-bug-0.1.0.tar.gz
Processing ./.tox/.tmp/package/3/tox-bug-0.1.0.tar.gz
  Installing build dependencies: started
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: tox-bug
  Building wheel for tox-bug (PEP 517): started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: tox-bug
  Building wheel for tox-bug (PEP 517): started
  Building wheel for tox-bug (PEP 517): finished with status 'done'
  Created wheel for tox-bug: filename=tox_bug-0.1.0-py3-none-any.whl size=1020 sha256=5d24828056fb6b4f155f98923bf14c8554f9d3d4c2052715bcb1659f3159c5c9
  Stored in directory: $HOME/.cache/pip/wheels/f9/ae/4b/2c7c9f9c392910b2d8ec6b2b88f7635c0ac650e0e0e6ca91ce
Successfully built tox-bug
  Building wheel for tox-bug (PEP 517): finished with status 'done'
  Created wheel for tox-bug: filename=tox_bug-0.1.0-py3-none-any.whl size=1020 sha256=5d24828056fb6b4f155f98923bf14c8554f9d3d4c2052715bcb1659f3159c5c9
  Stored in directory: $HOME/.cache/pip/wheels/fc/c0/15/202d2fd10458d08b491bb9b0079559a5370d05c3940d9f27a3
Successfully built tox-bug
Installing collected packages: tox-bug
Installing collected packages: tox-bug
Successfully installed tox-bug-0.1.0
Successfully installed tox-bug-0.1.0
isort finish: installpkg /tmp/tox-bug/.tox/.tmp/package/3/tox-bug-0.1.0.tar.gz after 8.14 seconds
isort start: envreport 
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697438] /tmp/tox-bug$ /tmp/tox-bug/.tox/isort/bin/python -m pip freeze >.tox/isort/log/isort-1.log
mypy finish: installpkg /tmp/tox-bug/.tox/.tmp/package/2/tox-bug-0.1.0.tar.gz after 8.20 seconds
mypy start: envreport 
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697439] /tmp/tox-bug$ /tmp/tox-bug/.tox/mypy/bin/python -m pip freeze >.tox/mypy/log/mypy-1.log
isort finish: envreport  after 0.33 seconds
isort installed: tox-bug @ file:///tmp/tox-bug/.tox/.tmp/package/3/tox-bug-0.1.0.tar.gz
isort start: run-test-pre 
isort run-test-pre: PYTHONHASHSEED='1386134793'
isort finish: run-test-pre  after 0.00 seconds
isort start: run-test 
isort run-test: commands[0] | poetry install -v
setting PATH=/tmp/tox-bug/.tox/isort/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697463] /tmp/tox-bug$ $HOME/.miniconda3/envs/tox-bug/bin/poetry install -v
mypy finish: envreport  after 0.31 seconds
mypy installed: tox-bug @ file:///tmp/tox-bug/.tox/.tmp/package/2/tox-bug-0.1.0.tar.gz
mypy start: run-test-pre 
mypy run-test-pre: PYTHONHASHSEED='1195468548'
mypy finish: run-test-pre  after 0.00 seconds
mypy start: run-test 
mypy run-test: commands[0] | poetry install -v
setting PATH=/tmp/tox-bug/.tox/mypy/bin:$HOME/.miniconda3/envs/tox-bug/bin:$HOME/.miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
[697464] /tmp/tox-bug$ $HOME/.miniconda3/envs/tox-bug/bin/poetry install -v
Using virtualenv: /tmp/tox-bug/.tox/mypy
Using virtualenv: /tmp/tox-bug/.tox/isort
Installing dependencies from lock file
Installing dependencies from lock file


Package operations: 17 installs, 0 updates, 0 removals, 2 skipped

  - Installing zipp (3.1.0)


Package operations: 17 installs, 0 updates, 0 removals, 2 skipped

  - Installing zipp (3.1.0)
  - Installing appdirs (1.4.4)
  - Installing appdirs (1.4.4)
  - Installing distlib (0.3.1)
  - Installing distlib (0.3.1)
  - Installing filelock (3.0.12)
  - Installing filelock (3.0.12)
  - Installing importlib-metadata (1.7.0)
  - Installing importlib-metadata (1.7.0)
  - Installing pyparsing (2.4.7)
  - Installing pyparsing (2.4.7)
  - Installing six (1.15.0)
  - Installing six (1.15.0)
  - Skipping atomicwrites (1.4.0) Not needed for the current environment
  - Installing attrs (20.1.0)
  - Skipping atomicwrites (1.4.0) Not needed for the current environment
  - Installing attrs (20.1.0)
  - Skipping colorama (0.4.3) Not needed for the current environment
  - Installing more-itertools (8.4.0)
  - Skipping colorama (0.4.3) Not needed for the current environment
  - Installing more-itertools (8.4.0)
  - Installing packaging (20.4)
  - Installing packaging (20.4)
  - Installing pluggy (0.13.1)
  - Installing pluggy (0.13.1)
  - Installing py (1.9.0)
  - Installing py (1.9.0)
  - Installing toml (0.10.1)
  - Installing toml (0.10.1)
  - Installing virtualenv (20.0.31)
  - Installing virtualenv (20.0.31)
  - Installing wcwidth (0.2.5)
  - Installing wcwidth (0.2.5)
  - Installing pytest (5.4.3)
  - Installing pytest (5.4.3)
  - Installing tox (3.19.0)
  - Installing tox (3.19.0)
  - Installing tox-bug (0.1.0)
  - Installing tox-bug (0.1.0)

[FileNotFoundError]
[Errno 2] No such file or directory: '/tmp/tox-bug/pyproject.toml'

Traceback (most recent call last):
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/clikit/console_application.py", line 131, in run
    status_code = command.handle(parsed_args, io)
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/clikit/api/command/command.py", line 120, in handle
    status_code = self._do_handle(args, io)
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/clikit/api/command/command.py", line 171, in _do_handle
    return getattr(handler, handler_method)(args, io, self)
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/cleo/commands/command.py", line 92, in wrap_handle
    return self.handle()
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/poetry/console/commands/install.py", line 94, in handle
    builder.build()
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/poetry/masonry/builders/editable.py", line 17, in build
    return self._setup_build()
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/site-packages/poetry/masonry/builders/editable.py", line 38, in _setup_build
    str(self._poetry.file), str(self._poetry.file.with_suffix('.tmp'))
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/shutil.py", line 571, in move
    copy_function(src, real_dst)
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/shutil.py", line 257, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "$HOME/.miniconda3/envs/tox-bug/lib/python3.7/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:

ERROR: InvocationError for command $HOME/.miniconda3/envs/tox-bug/bin/poetry install -v (exited with code 1)
mypy finish: run-test  after 12.46 seconds
mypy start: run-test-post 
mypy finish: run-test-post  after 0.00 seconds
cleanup /tmp/tox-bug/.tox/.tmp/package/2/tox-bug-0.1.0.tar.gz
ERROR: invocation failed (exit code 1)
✖ FAIL mypy in 21.515 seconds
isort finish: run-test  after 13.36 seconds
isort start: run-test-post 
isort finish: run-test-post  after 0.00 seconds
cleanup /tmp/tox-bug/.tox/.tmp/package/3/tox-bug-0.1.0.tar.gz
isort finish: parallel isort  after 22.39 seconds
✔ OK isort in 22.397 seconds
___________________________________ summary ____________________________________
ERROR:   mypy: parallel child exit code 1
  isort: commands succeeded
cleanup /tmp/tox-bug/.tox/.tmp/package/1/tox-bug-0.1.0.tar.gz
```

**Comment by gaborbernat** [comment]
Can't say I understand why this is happening, but if you figure it out feel free to open a PR for it.

**Comment by GigiusB** [comment]
I believe poetry renames the pyproject.toml during "poetry install". When tox runs it in parallel it may occur that one thread has renamed it while another tries to access it.
See also https://github.com/python-poetry/poetry/issues/1204

**Comment by MaxG87** [comment]
I will follow the discussion on Poetry's side than. Thanks for the effort.
